app/src/main/java/com/keylesspalace/tusky/util/EmojiCompatFont.kt

```diff
@@ -0,0 +1,363 @@
+     *
+     * @param context The Context
+     */
+    private fun getExistingFontFiles(context: Context): List<Pair<File, List<Int>>> {
+        // Only load it once
+        existingFontFileCache?.let {
+            return it
+        }
+            // If we call this on the system default font, just return nothing...
+            if (this === SYSTEM_DEFAULT) {
+                existingFontFileCache = emptyList()
+                return emptyList()
+            }
+
+            val directory = File(context.getExternalFilesDir(null), DIRECTORY)
+            // It will search for old versions using a regex that matches the font's name plus
+            // (if present) a version code. No version code will be regarded as version 0.
+            val fontRegex = "$name(\\d+(\\.\\d+)*)?\\.ttf".toPattern()
+            val ttfFilter = FilenameFilter { _, name: String -> name.endsWith(".ttf") }
+            val foundFontFiles = directory.listFiles(ttfFilter).orEmpty()
+            Log.d(TAG, String.format("loadExistingFontFiles: %d other font files found",
+                    foundFontFiles.size))
+
+            return foundFontFiles.map { file ->
+                val matcher = fontRegex.matcher(file.name)
+                val versionCode = if (matcher.matches()) {
+                    val version = matcher.group(1)
+                    getVersionCode(version)
+                } else {
+                    listOf(0)
+                }
+                Pair(file, versionCode)
+            }.sortedWith(
+                    Comparator<Pair<File, List<Int>>> { a, b -> compareVersions(a.second, b.second) }
+            ).also {
+                existingFontFileCache = it
+            }
+    }
+
+    /**
+     * Returns the current or latest version of this font file (if there is any)
+     *
+     * @param context The Context
+     * @return The file for this font with the current or (if not existent) highest version code or null if there is no file for this font.
+     */
+    private fun getLatestFontFile(context: Context): File? {
+        val current = getFontFile(context)
+        if (current != null && current.exists()) return current
+        val existingFontFiles = getExistingFontFiles(context)
+        return existingFontFiles.firstOrNull()?.first
+    }
+
+    private fun getVersionCode(version: String?): List<Int> {
+        if (version == null) return listOf(0)
+        return version.split(".").map {
+            it.toIntOrNull() ?: 0
+        }
+    }
+
+    fun downloadFontFile(context: Context,
+                         okHttpClient: OkHttpClient,
+                         onNext: Consumer<Float>,
+                         onError: Consumer<Throwable>,
+                         onComplete: Action) {
+        Observable.create { emitter: ObservableEmitter<Float> ->
+            // It is possible (and very likely) that the file does not exist yet
+            val downloadFile = getFontFile(context)!!
```

Comment:
tbh I'm surprised that Okio doesn't support any nio, at least I couldn't find anything.